### ISSUE TITLE
Python Build Changes in Broke Django App that used Mount Storage

### ISSUE THREAD

**Issue by getmarked** [Azure-App-Service/ImageBuilder#87]
Referencing [Python Build Changes](https://github.com/Azure-App-Service/KuduLite/wiki/Python-Build-Changes), this change broke my Django app.

We expected the Django App to run from `wwwroot` and used `BASE_DIR` to reference to our mounted blob storage. In Django, `BASE_DIR` points to `wwwroot`.

This change unexpectedly ran the app from a `tmp` directory instead causing the mounted blob storage not to be found.

This issue from experience by other user [here](https://github.com/microsoft/Oryx/issues/880)

Logging this issue here as a datapoint for the MS team.

**Comment by JennyLawrance** [comment]
@getmarked: Are you still blocked on this? were you able to override the BASE_DIR environment variable to leverage the $APP_ROOT path?

**Comment by getmarked** [comment]
In the end, we choose to hard code the path `home/site/wwwroot/MY_MOUNTED_STORAGE`.

If there was some way of knowing the changes before hand that would be nice. Our app went down for a while.

**Comment by JennyLawrance** [comment]
Agreed, that we should have communicated this change out sooner. This, this has been a learning experience for us.
We have sent out communications 12hrs ago, and are also evaluating how we can minimize the effect of this change. Will post an update on this thread in the next 6hrs with the next steps on our side.

**Comment by JennyLawrance** [comment]
Update: We are working on a mechanism to roll-back the changes.

**Comment by getmarked** [comment]
Thanks for attending to this matter :)

**Comment by geciciserkan** [comment]
Hey mates, our production has been effected from this problem. We have been trying to understand reason of the problem and we saw this post from App Service Advisor. Please fix this as soon as possible

**Comment by michaelteo** [comment]
> @getmarked Agreed, that we should have communicated this change out sooner. This, this has been a learning experience for us.
> We have sent out communications 12hrs ago, and are also evaluating how we can minimize the effect of this change. Will post an update on this thread in the next 6hrs with the next steps on our side.

Where are the communications for changes done? I was not able to know this page until someone replied on my issue on the Oryx github. 

Even the official MS Azure Support team don't seem to be aware of these changes when we raised the case.

**Comment by JennyLawrance** [comment]
Hi @michaelteo, We sent email and Azure UX Notifications to Python apps running on App service Linux. Are you saying you didn't receive this? I'm certainly curious on the cause, please reply with your subscription/app name to jennylaw(at)microsoft(dot)com and we can investigate.

**Comment by divinedeveloper** [comment]
Hi @JennyLawrance we too are facing this one. Somehow it works for us by using azure devops  but doesn't work when we use the deployment center from portal.

Please fix this ASAP and let us know. Thanks

**Comment by divinedeveloper** [comment]
@JennyLawrance today we tried deployment using azure-cli and azure devops and we faced the same /tmp/xxxxxxx directory issue.

**Comment by geciciserkan** [comment]
All django apps that runs on Azure App service had failure after first deployment due to this change. How is that possible Microsoft is not aware of that problem ?

**Comment by bw984** [comment]
This problem is also affecting python Flask apps that maintain local reference files.  Since the apps are now deployed out of a tmp folder instead of the home folder, the post deployment reference files are deleted each time the app is restarted.    Is the plan to revert the build process back to the home folder or do we need to modify our apps to fit into this new process?

**Comment by sanchitmehta** [comment]
@geciciserkan @bw984 @divinedeveloper We are in the process of reverting to the old deployment behavior across all regions. The new behavior would be an opt-in after this fix, we would send out notifications and give ample time to make changes when we change the default behavior.

You could pin to the old version by adding an app setting: `KUDU_BUILD_VERSION=0.0.1` -> This would ensure the build behavior remains the same even when we change default behavior after the notification.

To try the new version, you can add the app setting: `KUDU_BUILD_VERSION=1.0.0`.

Please share your app names directly or [indirectly](https://github.com/projectkudu/kudu/wiki/Reporting-your-site-name-without-posting-it-publicly) if you are still experiencing the problem and we can investigate further

**Comment by birketts-gavrielo** [comment]
@sanchitmehta Our deployment is deployed at tmp too. The location is https://bhive.birketts.co.uk/ or http://birkettsbhive1.azurewebsites.net/ . 
Is there any way to fix this so it deploys to wwwroot? I'm new with azure so not sure where to set the app setting on azure.

**Comment by bw984** [comment]
Python apps are (maybe still) deploying to tmp directories.  Please let us deploy to the home directory on our servers.

**Comment by sumitkute** [comment]
I think this is old thread and still open. Is this fixed other than setting the confinuration to KUDU_BUILD_VERSION=0.0.1

**Comment by AtitBimali** [comment]
The issue persists. One potential solution is to address it by adding `KUDU_BUILD_VERSION` with a value of `0.0.1` to your app service configuration.
